contracts/interfaces/New.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import "../core/Allo.sol";
+import "../interfaces/IStrategy.sol";
+
+abstract contract BaseStrategy is IStrategy {
+    Allo public immutable allo;
+
+    bytes32 public identityId;
+    uint256 public poolId;
+
+    constructor(address _allo) {
+        allo = Allo(_allo);
+    }
+
+    function initialize(
+        bytes32 _identityId,
+        uint256 _poolId,
+        bytes memory _recipientEligibilityData
+        bytes memory _voterEligibilityData
+        bytes memory _votingData
+        bytes memory _allocationData
+        bytes memory _distributionData
+    ) external {
+        require(msg.sender == address(allo), "only allo");
+        require(_identityId != bytes32(0), "invalid identity id"); // is it the case that identity id will never be 0? any reason it'd be better to use pool?
+        require(identityId == bytes32(0), "already initialized");
+        identityId = _identityId;
+        poolId = _poolId;
+
+        initializeRecipientEligibilityModule(_recipientEligibilityData);
+        initializeVoterEligibilityModule(_voterEligibilityData);
+        initializeVotingModule(_votingData);
+        initializeAllocationModule(_allocationData);
+        initializeDistributionModule(_distributionData);
+    }
+
+    function poolFunded(uint256 _amount) external;
+    // plus maybe some modifiers
+}
+
+interface IRecipientEligibilityModule {
+    enum RecipientStatus { None, Pending, Accepted, Rejected }
+    function initializeRecipientEligibilityModule(bytes _data) external;
+    function registerRecipients(bytes memory _data, address _sender) external payable returns (address);
+    function getRecipientStatus(address _recipientId) public view returns (RecipientStatus);
+    function allRecipients() public view returns (address[] memory);
+}
+
+interface IVoterEligibilityModule {
+    function initializeVoterEligibilityModule(bytes _data) external;
+    function isValidVoter(address _voter) public view returns (bool); // maybe should be renamed allocator, unless we change to vote below?
+}
+
+interface IVotingModule {
+    function initializeVotingModule(bytes _data) external;
+    function allocate(bytes memory _data, address _sender) external payable; //  is this the right name given module changes?
```

Comment:
I was thinking the same thing @zobront — now that we've gone through this exercise I think we can change this back to `vote()`

Reply:
I think so too.